src/toolkits/style_transfer/style_transfer_model_definition.cpp

```diff
@@ -0,0 +1,1008 @@
+
+  nn_spec->add_relu(
+      /* name */ "transformer_activation4",
+      /* input */ "transformer_instancenorm4__fwd");
+
+  nn_spec->add_padding(
+      /* name */ "transformer_pad5",
+      /* input */ "transformer_activation4",
+      /* padding_top */ 4,
+      /* padding_bottom */ 4,
+      /* padding_left */ 4,
+      /* padding_right */ 4,
+      /* padding_right */ padding_type::REFLECTIVE);
+
+  nn_spec->add_convolution(
+      /* name */ "transformer_conv5",
+      /* input */ "transformer_pad5",
+      /* num_output_channels */ 3,
+      /* num_kernel_channels */ 32,
+      /* kernel_height */ 9,
+      /* kernel_width */ 9,
+      /* stride_height */ 1,
+      /* stride_width */ 1,
+      /* padding */ padding_type::VALID,
+      /* weight_init_fn */ zero_weight_initializer());
+
+  nn_spec->add_inner_product(
+      /* name */ "transformer_instancenorm5_gamma",
+      /* input */ "index",
+      /* num_output_channels */ 3,
+      /* num_input_channels */ num_styles,
+      /* weight_init_fn */ ones_weight_initializer(),
+      /* bias_init_fn */ zero_weight_initializer());
+
+  nn_spec->add_inner_product(
+      /* name */ "transformer_instancenorm5_beta",
+      /* input */ "index",
+      /* num_output_channels */ 3,
+      /* num_input_channels */ num_styles,
+      /* weight_init_fn */ zero_weight_initializer(),
+      /* bias_init_fn */ zero_weight_initializer());
+
+  nn_spec->add_instancenorm(
+      /* name */ "transformer_instancenorm5__fwd_bn_",
+      /* input */ "transformer_conv5",
+      /* num_channels */ 3,
+      /* epsilon */ 1e-5);
+
+  nn_spec->add_multiplication(
+      /* name */ "transformer_instancenorm5__fwd_mult_gamma",
+      /* inputs */ {"transformer_instancenorm5__fwd_bn_",
+                    "transformer_instancenorm5_gamma"});
+
+  nn_spec->add_addition(
+      /* name */ "transformer_instancenorm5__fwd",
+      /* inputs */ {"transformer_instancenorm5__fwd_mult_gamma",
+                    "transformer_instancenorm5_beta"});
+
+  nn_spec->add_sigmoid(
+      /* name */ "transformer_activation5",
+      /* input */ "transformer_instancenorm5__fwd");
+
+  nn_spec->add_scale(
+      /* name */ "stylizedImage",
+      /* input */ "transformer_activation5",
+      /* shape_c_h_w */ {1},
+      /* weight_init_fn */ const_weight_initializer(255.0));
+}
+
+void load_weights(std::unique_ptr<model_spec>& nn_spec, std::string& path) {
+  std::unique_ptr<model_spec> weight_spec(new model_spec(path));
+  float_array_map nn_params = weight_spec->export_params_view();
+
+  for (auto iter = nn_params.begin(); iter != nn_params.end();) {
+    if (iter->first.find("inst") != std::string::npos) {
+      nn_params.erase(iter++);
+      continue;
+    }
+    ++iter;
+  }
+
+  nn_spec->update_params(nn_params);
+}
+
+}  // namespace
+
+std::unique_ptr<model_spec> init_resnet(std::string& path) {
+  std::unique_ptr<model_spec> spec(new model_spec(path));
+  return spec;
+}
```

Comment:
Honest question: what do you see as the benefit to defining this function (which has the same implementation as `init_vgg_16` below)?

Reply:
It's for api, readability. The init_resnet has many overloaded functions and to keep it consistent I added both function. I could refactor to just init_model and call it a day there.

Reply:
I wonder if we could improve the design here, but what we have is fine for now. We can always refactor it later if desired